src/time-range-picker/time-range-picker.tsx

```diff
@@ -31,30 +29,23 @@ export const TimeRangePicker = ({
         useState<boolean>(false);
     const [showEndTimeSelector, setShowEndTimeSelector] =
         useState<boolean>(false);
-    const [startTimeVal, setStartTimeVal] = useState("");
-    const [endTimeVal, setEndTimeVal] = useState("");
+    const [startTimeVal, setStartTimeVal] = useState<string>("");
+    const [endTimeVal, setEndTimeVal] = useState<string>("");
 
     const nodeRef = useRef<HTMLDivElement>();
 
     // =============================================================================
     // EFFECTS
     // =============================================================================
-
     useEffect(() => {
         if (value) {
-            setStartTimeVal(value.startTime);
-            setEndTimeVal(value.endTime);
+            setStartTimeVal(value.start);
+            setEndTimeVal(value.end);
         }
     }, []);
-    useEffect(() => {
-        document.addEventListener("mousedown", handleMouseDownEvent);
-        document.addEventListener("keyup", handleKeyUpEvent);
 
-        return () => {
-            document.removeEventListener("mousedown", handleMouseDownEvent);
-            document.removeEventListener("keyup", handleKeyUpEvent);
-        };
-    }, [showEndTimeSelector]);
```

Comment:
missed out during review. these callbacks have stale state. have also amended similar issues in Timepicker